### PR TITLE
Add trash shortcut in toolbar for for incognito tabs

### PR DESCRIPTION
**Pull request by jedireza**
Currently in incognito tabs, we disable the spaces bookmark button. In this PR I'm proposing we utilize that space to allow users to quickly discard and close the tab.

### How was this tested?
- [x] I tested this manually

### Manual test cases
I tested in the simulator.

#### iPhone

https://user-images.githubusercontent.com/979929/170113229-807d785e-c247-4ae5-abce-7aa732a205ce.mp4

#### iPad

https://user-images.githubusercontent.com/979929/170142902-956e2e57-ba76-4247-92f5-3eaac029dfa2.mp4

## Issues addressed
N/A

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift [hunk 1]
```diff
@@ -47,6 +48,17 @@ extension BrowserViewController: ToolbarDelegate {
...
+            case .closeTab:
+                ClientLogger.shared.logCounter(
+                    .ClickClose,
+                    attributes: EnvironmentHelper.shared.getAttributes() + [toolbarActionAttribute]
+                )
+
+                if let tab = self.tabManager.selectedTab {
+                    self.tabManager.close(tab)
+                    self.showTabTray()
+
```
Comment (evancrow) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+TabToolbarDelegate.swift:60]: nit: remove extra line

File: Client/Frontend/TabChrome/Toolbar/TabToolbarView.swift [hunk 2]
```diff
@@ -62,8 +63,14 @@ struct TabToolbarView: View {
                 identifier: "TabOverflowButton"
             )
             neevaButton
-            TabToolbarButtons.AddToSpace(
-                weight: .medium, action: { performAction(.addToSpace) })
+            if incognitoModel.isIncognito {
+                TabToolbarButtons.CloseTab(
+                    action: { performAction(.closeTab) })
+
```
Comment (evancrow) [inline Client/Frontend/TabChrome/Toolbar/TabToolbarView.swift:69]: nit: remove extra line

**Review by evancrow — APPROVED** [review]
LGTM - two nits

I wonder if this action should be protected at all? Maybe a double click, but I get that it's a quick burner action for incognito.

**Comment by jedireza** [comment]
Thanks for the review @evancrow 👍 

> I wonder if this action should be protected at all? Maybe a double click, but I get that it's a quick burner action for incognito.

Yep, quick burner is the idea. Also a random double tap icon would probably feel broken since nothing else works that way.

I'll hold off on this until we have time to discuss more broadly as a team.

**Comment by jedireza** [comment]
We're putting this on the back burner.

**Comment by jedireza** [comment]
I'm throwing this behind a feature flag.

**Review by darinwf — APPROVED** [review]
LGTM, thanks for sorting the feature flags!

**Pull request merged**